Added propdates documentation, refined escrow image assets (#2)
SmartInvoiceXYZ/nouns-builder-docs: src/content/docs/reference/attestations.mdx

---
title: Propdate EAS Schema
---

## Overview

This schema defines how users of Nouns Builder can submit *Propdates* to DAO proposals, i.e. *updates*, *comments*, and *milestone-related actions* to proposals using the [Ethereum Attestation Service (EAS)](https://docs.attest.sh).
All attestations using this schema are displayed under the **Propdates** section of the proposal UI.
It supports Markdown, IPFS links, structured JSON, and threaded discussions via reply references. All attestations are displayed under the *Propdates* tab in the proposal view.

## Schema Specification

> The schema with the UID `0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3` enables Nouns Builder Propdates. It is deployed on-chain on all networks supported by Nouns Builder, consisting of mandatory and optional fields. The payload displayed on the frontend is wrapped in `message`, which MAY contain a *JSON* string.

Reviewer: Could we elaborate on the schema here? 

I had written this on the issue which we could take from:

<h2>✅ Final EAS Schema (on-chain)</h2>
<p>This is the schema you'll register with EAS and use in Solidity:</p>
<pre><code class="language-solidity">bytes32 propId;
bytes32 replyToUID;
uint8 messageType;      // 0: INLINE_TEXT, 1: INLINE_JSON, 2: URL_TEXT, 3: URL_JSON
string message;         // Inline content or off-chain URL
</code></pre>
<ul>
<li>
<p><code inline="">propId</code>: ID of the proposal being commented on.</p>
</li>
<li>
<p><code inline="">replyToUID</code>: UID of another attestation being replied to (or 0x0 if top-level).</p>
</li>
<li>
<p><code inline="">messageType</code>: Determines how to interpret the <code inline="">message</code> field.</p>
</li>
<li>
<p><code inline="">message</code>: Inline content or a URL (e.g., IPFS) depending on <code inline="">messageType</code>.</p>
</li>
</ul>
<hr>
<h2>✅ Final <code inline="">message</code> JSON Schema (for INLINE_JSON or URL_JSON)</h2>
<p>This JSON will be embedded in <code inline="">message</code> (for INLINE_JSON) or hosted off-chain (for URL_JSON):</p>
<pre><code class="language-json">{
  "milestoneId": 2,                              // Optional
  "content": "Delivered frontend with tests.",   // Required
  "labels": ["delivery", "frontend"],            // Optional
  "attachments": ["ipfs://.../demo.mp4"]         // Optional
}
</code></pre>
<h3>Field Summary:</h3>

Field | Type | Description
-- | -- | --
milestoneId | number | Optional. Include only if this comment relates to a milestone.
content | string | Required. Main body of the update, always interpreted as Markdown.
labels | string[] | Optional tags (e.g. ["QA", "review"]).
attachments | string[] | Optional list of URLs (preferably IPFS) for media, files, or demos.

<ul>
<li>
<p>Markdown is always used for rendering <code inline="">content</code>, even if plain text is submitted.</p>
</li>
<li>
<p><code inline="">milestoneId</code> is embedded only when relevant, making the on-chain schema cleaner.</p>
</li>
</ul>

### Deployed Schemas

- [Base](https://base.easscan.org/schema/view/0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3),

> - [Zora](),

Reviewer: @benedictvscriticus we should remove zora from here and mention somewhere that propdates is not supported for zora and zora sepolia

| `content`     | `string`   | [**Required**] text content                  |
| `labels`      | `string[]` | Tags for filtering/search                    |
| `attachments` | `string[]` | List of media or file URLs (preferably IPFS) |

## Usage

### Example Attestation

[View on EASScan](https://base.easscan.org/attestation/view/0xf1c43e68c96c29831af1760979b3c7f52abfb14561434381a5983f356c2f9ab8)

```json
{
  "propId": "0x20dce33aa50446e11ad5e97321f50e98c5d4fa41b0cfa87b5a525a6d1100fc4b",
  "replyToUID": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "messageType": 1,
  "message": "{\"milestoneId\":1,\"content\":\"Milestone 1 complete. All Figma wireframes uploaded.\",\"labels\":[\"figma\",\"ux\"],\"attachments\":[\"ipfs://bafybeihpky3lq4m/demo.mp4\"]}"
}
```

### Workflow for non-UI Propdates

1. **Create attestation**  
   A wallet with authority can submit an attestation directly through [EAS](https://base.easscan.org/) using the schema UID.

2. **Fetch and display attestations**  
   Fetch the schema on the relevant network using EAS Scan and the Schema UID. Connect your wallet and post an update.

3. **Propdates versus release requests

> Release request must be made indepentently of updates and comments, using the `Release Milestone` functionality, as described in the [Nouns Builder Escrow Proposal Guide](./guides/builder-escrow-proposal.mdx)

Reviewer: ```suggestion
   Release request must be made indepentently of updates and comments, using the `Release Milestone` functionality, as described in the [Nouns Builder Escrow Proposal Guide](./guides/builder-escrow-proposal.mdx)
```

### Usage in Nouns Builder Frontend

- Appears in **Propdates** tab on proposal page.
- Posts by DAO members are shown by default but anyone can:
  - Post progress updates
  - Comment on milestones
  - Start threaded conversations

## Attestation Policy and Deploying Schemas

Only attestations from DAO members are displayed under **Propdates** by default, but users can select **All Propdates** to see non-member comments.

> The attestion schema can be deployed to additional networks using the same schema hash by anyone.

Reviewer: ```suggestion
The attestation schema can be deployed to additional networks using the same schema by anyone.
```

---
sidebar:
  order: 6
title: Propdate EAS Schema
---

> ## Overview

Reviewer 2: is this a dev doc?

Author: yes

Author: Why?

This schema defines how users of Nouns Builder can submit *Propdates*, i.e., *updates*, *comments*, and *milestone-related actions*, to DAO proposals using the [Ethereum Attestation Service (EAS)](https://docs.attest.org).
All attestations using this schema are displayed under the **Propdates** section of the proposal UI. It supports Markdown, IPFS links, structured JSON, and threaded discussions via reply references.

## Sequence Diagram

![EAS Attestaion Flow](./public/guides/propdateGuide/attestation-flow.png)

```mermaid
sequenceDiagram
    participant User
    participant DAO Frontend
    participant Ethereum Attestation Service (EAS)
    participant Blockchain

    User->>DAO Frontend: Compose Propdate update
    DAO Frontend->>User: Prompt to sign transaction
    User->>Blockchain: Sign and submit Propdate attestation
    Blockchain->>EAS: Store attestation
    DAO Frontend->>EAS: Fetch Propdates for proposal/activity feed
    EAS->>DAO Frontend: Return Propdate data
    DAO Frontend->>User: Display Propdates (filtered by member/all)
```

## Schema Specification

> The schema with the UID `0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3` enables Nouns Builder Propdates. It is deployed on-chain on all networks supported by Nouns Builder **except Zora**. The schema consists of mandatory and optional fields. The payload displayed on the frontend is wrapped in a `message` field, which MAY contain a *JSON* string.

Reviewer 2: onchain

Author: done

### Deployed Schemas

- [Base](https://base.easscan.org/schema/view/0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3),
- Zora Mainnet **(not supported)**
- [OP Mainnet](https://optimism.easscan.org/schema/view/0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3),
- [Ethereum Mainnet](https://easscan.org/schema/view/0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3),
- [Base Sepolia](https://base-sepolia.easscan.org/schema/view/0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3),
- [OP Sepolia](https://optimism-sepolia.easscan.org/schema/view/0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3),
- Zora Sepolia **(not supported)**
- [Ethereum Sepolia](https://sepolia.easscan.org/schema/view/0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3)

### Schema Structure

```solidity

> bytes32 propId;

Reviewer: ```suggestion
bytes32 proposalId;
```

> bytes32 replyToUID;

Reviewer: ```suggestion
bytes32 originalMessageId;
```

uint8 messageType;      // 0: INLINE_TEXT, 1: INLINE_JSON, 2: URL_TEXT, 3: URL_JSON
string message;         // Inline content or off-chain URL
```

### Field Breakdown

| Field               | Type      | Description |
|---------------------|-----------|-------------|

> | `propId`        | `bytes32` | Proposal ID (usually keccak256 of the proposal metadata or ID hash) |

Reviewer: ```suggestion
| `proposalId`        | `bytes32` | Proposal ID (usually keccak256 of the proposal metadata or ID hash) |
```

### Deployed Schemas

- [Base](https://base.easscan.org/schema/view/0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3),
- **EAS, hence Nouns Builder Propdates are not available on Zora Mainnet.**
- [OP Mainnet](https://optimism.easscan.org/schema/view/0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3),
- [Ethereum Mainnet](https://easscan.org/schema/view/0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3),
- [Base Sepolia](https://base-sepolia.easscan.org/schema/view/0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3),
- [OP Sepolia](https://optimism-sepolia.easscan.org/schema/view/0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3),
- **EAS, hence Nouns Builder Propdates are not available on Zora Sepolia.**
- [Ethereum Sepolia](https://sepolia.easscan.org/schema/view/0x8bd0d42901ce3cd9898dbea6ae2fbf1e796ef0923e7cbb0a1cecac2e42d47cb3)

### Schema Structure

> ```solidity

Reviewer: @benedictvscriticus if looks like you reverted the changes I had made to the schema. The name of the variables is important as it gets cached when deploying the schema on-chain.

It should be 

```solidity
bytes32 proposalId;
bytes32 originalMessageId;
uint8 messageType;  // 0: INLINE_TEXT, 1: INLINE_JSON, 2: URL_TEXT, 3: URL_JSON
string message;     // Inline content or off-chain URL
```

Reviewer: fixed in a11858d1719ddd16f4e5ece2590f6c45f4fbfa28

| Field               | Type      | Description |
|---------------------|-----------|-------------|

> | `propId`        | `bytes32` | Proposal ID (usually keccak256 of the proposal metadata or ID hash) |

Reviewer: this should be `proposalId`

Reviewer: fixed in a11858d1719ddd16f4e5ece2590f6c45f4fbfa28
